src/components/iconButton/IconButton.tsx

```diff
@@ -1,57 +1,89 @@
-import React, { useRef, useImperativeHandle, forwardRef } from 'react';
-import PropTypes from 'prop-types';
+import React, { useRef, useImperativeHandle, forwardRef, ReactNode } from 'react';
 import Box from '../box';
 import Icon from '../icon';
 import cx from 'classnames';
 import buttonTheme from '../button/theme.css';
 import theme from './theme.css';
 import LoadingSpinner from '../loadingSpinner';
+import { BoxProps } from '../box/Box';
+
+type SIZES = 'small' | 'medium' | 'large';
+type TINTS = 'lightest' | 'light' | 'normal' | 'dark' | 'darkest';
+type COLOURS = 'neutral' | 'white' | 'mint' | 'violet' | 'ruby' | 'gold' | 'aqua' | 'teal';
+
+export interface IconButtonProps extends Omit<BoxProps, 'ref'> {
```

Comment:
all props from here that are passed to `Box` (L 99) needs to be mentioned here, if you are somehow overriding them

Reply:
Missed those 😬 . Adding the overrided ones now.